frame/election-provider-support/src/lib.rs

```diff
@@ -391,7 +391,7 @@ pub trait ElectionProviderBase {
 				if desired_targets <= Self::MaxWinners::get() {
 					Ok(desired_targets)
 				} else {
-					Err("desired_targets should not be greater than MaxWinners")
+					Err("desired_targets must not be greater than MaxWinners.")
```

Comment:
I am leaning towards throwing an error if `desired_targets < MaxWinners`. Alternatively, we can cap the `desired_targets` to `MaxWinners` silently.

Reply:
My feeling is that error-ing the snapshot creation because `desired_targets < MaxWinners` is probably too harsh. I think that capping the `desired_targets` to `MaxWinners` would be a good option, but I'd do it when in `fn snapshot_creation_external()` when handling the error from this call. And add a warning too. Something along the lines of:

```rust
fn create_snapshot_external(
  // ...
  let mut desired_targets = <Pallet<T> as ElectionProviderBase>::desired_targets_checked().or_else(|e| 
     match e {
        Error::DesiredTargetsGreaterThanMaxWinners => {
             // warning: capping the desired targets
             Ok(<Pallet<T> as ElectionProviderBase>::MaxWinners::get())
          }
          _ => Err(ElectionError::DataProvider(e)),
  })?;

  // ...
```

Reply:
I have been split about this. We are already capping `desired_targets` to `electable_targets` so in that way it makes sense to cap `desired_targets` to `MaxWinners` as well. 

But to present another perspective why this is a different case, `electable_targets` is a dynamic value that depends on total number of validator intentions while `MaxWinners` and `desired_targets` are static value configured in the runtime. If we consider this as a configuration issue, we should ideally fail till this configuration is fixed. We also have `try_state` checks to make sure we are not making this configuration error before it goes live in production.

Reply:
`MaxWinners` is the cap that should NEVER be reached, I think returning an error is better.